app/src/main/java/org/jellyfin/androidtv/content/MediaContentProvider.kt

```diff
@@ -0,0 +1,171 @@
+	}
+
+	override fun query(uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?,
+					   sortOrder: String?): Cursor? {
+		Timber.d("Query: %s", uri)
+		// Use the UriMatcher to see what kind of query we have and format the db query accordingly
+		when (sURIMatcher.match(uri)) {
+			SEARCH_SUGGEST -> {
+				Timber.d("search suggest: ${selectionArgs?.get(0)} URI: $uri")
+				var limit = 10
+				if (uri.getQueryParameter("limit") != null)
+					limit = uri.getQueryParameter("limit")!!.toInt()
+				return uri.lastPathSegment?.let { getSuggestions(it.toLowerCase(), limit) }
+			}
+			else -> throw IllegalArgumentException("Unknown Uri: $uri")
+		}
+	}
+
+	/**
+	 * Gets the resumable items or returns null
+	 */
+	private suspend fun searchItems(query: String, qLimit: Int): BaseItemDtoQueryResult? {
+		// Get user or return if no user is found (not authenticated)
+		val userId = TvApp.getApplication().currentUser?.id?.toUUIDOrNull() ?: return null
+
+		return itemsApi.getItems(
+				recursive = true,
+				enableTotalRecordCount = true,
+				collapseBoxSetItems = false,
+				excludeLocationTypes = listOf(LocationType.VIRTUAL),
+				sortBy = listOf(ItemSortBy.DateCreated),
+				sortOrder = listOf(SortOrder.DESCENDING),
+				searchTerm = query,
+
+				userId = userId,
+				imageTypeLimit = 1,
+				limit = qLimit,
+				fields = listOf(ItemFields.DATE_CREATED, ItemFields.STUDIOS, ItemFields.GENRES, ItemFields.TAGLINES, ItemFields.PROVIDER_IDS, ItemFields.MEDIA_STREAMS)
+		).content
+	}
+
+	private fun getSuggestions(query: String, limit: Int): Cursor = runBlocking {
+		val searchResult = async { searchItems(query, limit) }.await()
+		if (searchResult != null) {
+			Timber.d("Search returned %d items", searchResult.totalRecordCount)
+		}
+
+		val menuCols = arrayOf(
+				BaseColumns._ID,
+				SearchManager.SUGGEST_COLUMN_TEXT_1,
+				SearchManager.SUGGEST_COLUMN_TEXT_2,
+				SearchManager.SUGGEST_COLUMN_ICON_1,
+				SearchManager.SUGGEST_COLUMN_RESULT_CARD_IMAGE,
+				SearchManager.SUGGEST_COLUMN_INTENT_ACTION,
+				SearchManager.SUGGEST_COLUMN_INTENT_DATA,
+				SearchManager.SUGGEST_COLUMN_VIDEO_WIDTH,
+				SearchManager.SUGGEST_COLUMN_VIDEO_HEIGHT,
+				SearchManager.SUGGEST_COLUMN_PRODUCTION_YEAR,
+				SearchManager.SUGGEST_COLUMN_DURATION,
+				SearchManager.SUGGEST_COLUMN_SHORTCUT_ID
+		)
+		val mc = MatrixCursor(menuCols)
+
+		searchResult?.items?.forEach { item ->
+			val imageUri = Uri.parse(imageApi.getItemImageUrl(
+					itemId = item.id,
+					imageType = ImageType.PRIMARY,
+					format = ImageFormat.PNG,
+					maxHeight = 512,
+					maxWidth = 512
+			))
+			val duration = when {
+				item.runTimeTicks != null -> (item.runTimeTicks!! / TICKS_IN_MILLISECOND).toInt()
+				else -> 0
+			}
+			val release = when {
+				item.premiereDate != null -> item.premiereDate!!.year.toString()
+				else -> ""
+			}
+
+			Timber.d("Result %s/%s - %d / %s", item.name, item.id, duration, release)
+			mc.addRow(arrayOf<Any>(
```

Comment:
```suggestion
			mc.addRow(arrayOf(
```

Reply:
Can you add comments to all parameters like `// SUGGEST_COLUMN_TEXT_1` etc. so it's easier to read which columns the values are for?